### ISSUE TITLE
"Parameter comment not aligned correctly" for multi-line comments

### ISSUE THREAD

**Issue by cweiske** [squizlabs/PHP_CodeSniffer#1467]
I have the following parameter comment:

```
/**
 * Calculate permissions.
 *
 * Hook for BackendUserAuthentication::calcPerms()
 *
 * @param array  $params      Keys:
 *                            - row: page row
 *                            - outputPermissions: Currently calculated
 *                              permissions
 * @param object $backendUser Current backend user object
 *
 * @return integer Permissions for the given page
 */
public function calcPerms($params, BackendUserAuthentication $backendUser)
{
```

PHP_CodeSniffer 3.0.0 complains that the "permissions" are not aligned correctly:
```
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 82 | ERROR | [x] Parameter comment not aligned correctly; expected
    |       |     28 spaces but found 30
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

That worked fine with phpcs 2.x

I don't see another way to document array keys with long texts.

**Comment by gsherwood** [comment]
Yeah, that new check is probably a bit aggressive at the moment.

You can exclude it from your ruleset using `<exclude name="PEAR.Commenting.FileComment.ParamCommentAlignment" />` if you'd like, but I'll also try and change it to enforce exact indent on the first line and then use a minimum indent for subsequent lines.

**Comment by gsherwood** [comment]
I just realised what this will actually break. It's comments like this:
```php
/**
 * Processes the test.
 *
 * @param int $phpcsFile The PHP_CodeSniffer
 *                   file where the
 *                                        token occurred.
 * @param int $stackPtr  The position in the tokens stack
 *                                    where the listening token type
 *                                           was found.
 *
 * @return void
 * @see    register()
 */
function process(File $phpcsFile, $stackPtr)
{

}//end process()
```
The subsequent lines are clearly indented too far and the sniff should report an error and fix them, but it can't tell the difference between this case and the report cases, where the extra 2 spaces are intentional.

All I can probably do here is turn this error into two different error codes (ParamCommentAlignment and ParamCommentAlignmentExceeded maybe) so that you can exclude the case where too much space is used without excluding the case where too little is used.

**Comment by gsherwood** [comment]
I've now added the second error code so you can exclude comment lines with excessive padding (`ParamCommentAlignmentExceeded`) while still producing errors for comment line with too little padding (`ParamCommentAlignment`).
